### ISSUE TITLE
Please provide some more information on debugging code hosted on a local web server

### ISSUE THREAD

**Issue by oskaremil** [MicrosoftDocs/vsts-docs#4334]
Quote: "In order to debug the extension using Visual Studio or Browser Developer Tools and speed up the development without re-deploying extension each time you change source code, you need change manifest adding baseUri property

This tells Azure DevOps Services to load the extension from your local web server instance (e.g. IISExpress in Visual Studio). After changing manifest you need to deploy and install this debugging extension only once."

- I understand, while I am in a debugging/development process I can change the code as much as I like locally and see the changes immediately on Azure DevOps

- I don't understand: What do I do when I am ready to publish a new version ? 
-- Do I remove the baseUri property before I publish a new version and re-add it and publish a new version the next time I am doing development on the extension ?
-- When the extension is in debug mode, what will happen to the other DevOps users in the same organization ? They don't have any localhost webserver serving the files. Will the extension be unavailable for them, worst case break something on the page, when I upload the debugger extension with the baseUri property set ? 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 14955ee9-29f2-46fe-3a8d-a42666f53ffc
* Version Independent ID: 5d417cd0-a671-ee30-8fca-5f8b0addd328
* Content: [Develop a web extension for Azure DevOps Services](https://docs.microsoft.com/en-us/azure/devops/extend/get-started/node?view=azure-devops#feedback)
* Content Source: [docs/extend/get-started/node.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/extend/get-started/node.md)
* Product: **devops**
* Technology: **devops-ecosystem**
* GitHub Login: @willsmythe
* Microsoft Alias: **wismythe**

**Comment by DalpheSDC** [comment]
Please also provide how to debug extension on-premise servers

**Comment by dwhathaway** [comment]
Yes, you just remove the baseUri property for your "release" extension.  There's a couple of approaches to this, and it depends on how your team wants to work.  

1. You can use an override file when building the extension vsix for dev, and just append the baseUri property (that's the process we followed in that blog post that @tiagonmas shared).
2. Keep a separate vss-extension.json manifest file specifically for dev.  You might want to do this if you have more extensive changes you want to make for dev (e.g. loading different image files, having additional dev-specific hubs, etc.).

As a best practice, you should change the `id` parameter (see link below) in your dev extension to something different from the release one.  This will allow you to host them side-by-side, and do breakpoint developing while not affecting the "production" version.

https://github.com/microsoft/azure-devops-extension-hot-reload-and-debug/blob/38ae308fe72992500a44323611d35f847480b8d5/configs/dev.json#L2

**Comment by dwhathaway** [comment]
@dalphesdc AFAIK, this same process should "just work" with on-prem servers the same way as it does for hosted.  The process to install the extension is a bit different, but it's the same premise - loading a localhost URL from inside a browser page.  Here's the documentation for installing extensions from a local gallery - https://docs.microsoft.com/en-us/azure/devops/marketplace/get-tfs-extensions?view=azure-devops-2019#install-azure-devops-server-extensions-from-the-local-gallery

**Comment by justhchung** [comment]
Thank you for your feedback. 
We are closing this issue as there is no action to take on our part to address the issue.  The guidance provided is correct, requested additional information was not provided, or the content referenced is no longer relevant.
